run_all_tools.sh

```diff
@@ -97,62 +103,110 @@ function run_cloudrail {
 }
 
 function run_all {
-  run_checkov
-  run_cloudrail
-  run_kics
-  run_snyk
-  run_terrascan
-  run_tfsec
+  echo Now running Checkov on all cases
+  run_checkov $1
+  echo Now running Cloudrail on all cases
+  run_cloudrail $1
+  echo Now running KICS on all cases
+  run_kics $1
+  echo Now running Snyk on all cases
+  run_snyk $1
+  echo Now running Terrascan on all cases
+  run_terrascan $1
+  echo Now running TFsec on all cases
+  run_tfsec $1
 }
 
-# Set up AWS access for plan
-if [ -z "$AWS_ACCESS_KEY_ID" ]; then
-  echo "To run this script, you'll need AWS credentials (for use with terraform plan)."
-  exit 1
-fi
-export AWS_REGION=us-west-1
+if [[ $1 == *"all" ]] || [[ $1 == *"cloudrail" ]]; then
+  # Set up AWS access for plan
+  if [ -z "$AWS_ACCESS_KEY_ID" ]; then
+    echo "To run this script, you'll need AWS credentials (for use with terraform plan)."
+    exit 1
+  fi
+  export AWS_REGION=us-west-1
 
-# Generate all plan files
-echo Generating plan files, where they do not exist yet
+  # Generate all plan files
+  echo Generating plan files, where they do not exist yet
 
-find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do
-  echo $test_case
-  ORG_PATH=$PWD
-  cd $test_case
-  if [ ! -f plan.out ]; then
-    terraform init
-    terraform plan -out=plan.out
-  fi
-  cd $ORG_PATH
-done
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do
+    echo $test_case
+    ORG_PATH=$PWD
+    cd $test_case
+    if [ ! -f plan.out ]; then
+      terraform init
+      terraform plan -out=plan.out
+    fi
+    cd $ORG_PATH
+  done
+fi
 
 case $1 in
 run_checkov)
```

Comment:
I think it'd be better to have a parameter for this script, like `--rerun` where it will use `read_rerun` instead of `find_all_test_cases`. That way this `case` will be cleaner, and not duplicated for each tool.

BTW - the `rerun` capability is a great idea :)

Reply:
Sure.